### PR TITLE
Bluetooth: Skip address setting for passive scan

### LINKED ISSUE #22088: Bluetooth Mesh friendship is cleared due to no Friend response reception
**Describe the bug**
In the friendship established between a Friend Node (FN) and a Low Power Node (LPN) in a Bluetooth Mesh, the LPN generates and sends the Friend Poll message, which is correctly received by the FN. However, after that, the LPN is not able to start the scanning, with the following debug output:
[00:00:32.176,910] <err> bt_mesh_adv: starting scan failed (err -5)   
This produces the FN response is not received, and after some retries the friendship is cleared.

**To Reproduce**
1. Establish a Friendiship between a LPN and a FN, using default configuration
2. After the Friend Poll, the LPN is not able to scan 
3. Output shows <err> bt_mesh_adv: starting scan failed (err -5)  
4. After the timeout expires, the friendship is cleared

**Expected behavior**
The LPN receives the Friend response 

**Screenshots or console output**
Debugging output from LPN using CONFIG_BT_MESH_DEBUG_LOW_POWER and CONFIG_BT_MESH_DEBUG_ADV options:
```
Initializing...                                                                                                                                  
Unicast address: 0x0003                                                                                                                          
Bluetooth initialized                                                                                                                            
Mesh initialized                                                                                                                                 
Loading stored settings                                                                                                                          
Provisioning completed                                                                                                                           
Configuring...                                                                                                                                   
Publishing heartbeat messages                                                                                                                    
Configuration complete                                                                                                                           
[00:00:00.011,169] <inf> fs_nvs: 8 Sectors of 4096 bytes                                                                                         
[00:00:00.011,169] <inf> fs_nvs: alloc wra: 0, ff0                                                                                               
[00:00:00.011,169] <inf> fs_nvs: data wra: 0, 0                                                                                                  
[00:00:00.012,512] <inf> bt_hci_core: HW Platform: Nordic Semiconductor (0x0002)                                                                 
[00:00:00.012,542] <inf> bt_hci_core: HW Variant: nRF52x (0x0002)                                                                                
--- 9 messages dropped ---                                                                                                                       
[00:00:00.024,627] <dbg> bt_mesh_lpn.bt_mesh_lpn_init:                                                                                           
[00:00:00.024,627] <dbg> bt_mesh_adv.bt_mesh_scan_enable:                                                                                        
[00:00:00.024,902] <dbg> bt_mesh_lpn.bt_mesh_lpn_init: Waiting 1000 ms for messages                                                              
[00:00:00.024,902] <dbg> bt_mesh_lpn.lpn_set_state: disabled -> timer                                                                            
[00:00:00.067,260] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.069,091] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.069,549] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.069,976] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.070,190] <dbg> bt_mesh_lpn.bt_mesh_lpn_group_add: group 0xc000                                                                         
[00:00:00.070,434] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.070,892] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.071,044] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.071,289] <wrn> bt_mesh_cfg_cli: Unexpected App Key Status message                                                                      
[00:00:00.071,289] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.071,533] <wrn> bt_mesh_cfg_cli: Unexpected Model App Status message                                                                    
[00:00:00.071,533] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.071,746] <wrn> bt_mesh_cfg_cli: Unexpected Model App Status message                                                                    
[00:00:00.071,777] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.072,021] <wrn> bt_mesh_cfg_cli: Unexpected Model Subscription Status message                                                           
[00:00:00.072,021] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.072,265] <wrn> bt_mesh_cfg_cli: Unexpected Heartbeat Publication Status message                                                        
[00:00:00.072,296] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.072,601] <wrn> bt_mesh_transport: Ignoring heartbeat to non-subscribed destination                                                     
[00:00:00.072,631] <dbg> bt_mesh_lpn.bt_mesh_lpn_msg_received: Restarting establishment timer                                                    
[00:00:00.072,784] <dbg> bt_mesh_adv.bt_mesh_adv_send: type 0x01 len 21: 373fa347ad8cf0510fa9e4e308e7016~                                        
[00:00:00.072,875] <dbg> bt_mesh_adv.adv_send: type 1 len 21: 373fa347ad8cf0510fa9e4e308e7016~                                                   
[00:00:00.072,875] <dbg> bt_mesh_adv.adv_send: count 3 interval 20ms duration 120ms                                                              
[00:00:00.073,577] <dbg> bt_mesh_adv.adv_send: Advertising started. Sleeping 120 ms                                                              
[00:00:00.193,786] <dbg> bt_mesh_adv.adv_send: Advertising stopped                                                                               
[00:00:01.072,692] <dbg> bt_mesh_lpn.lpn_timeout: state: timer                                                                                   
[00:00:01.072,692] <dbg> bt_mesh_lpn.lpn_timeout: Starting to look for Friend nodes                                                              
[00:00:01.072,723] <dbg> bt_mesh_lpn.lpn_set_state: timer -> enabled                                                                             
[00:00:01.072,723] <dbg> bt_mesh_lpn.send_friend_req:                                                                                            
[00:00:01.072,875] <dbg> bt_mesh_adv.bt_mesh_adv_send: type 0x01 len 28: 37d544ee68f31aa29525ffd3c210726~                                        
[00:00:01.072,937] <dbg> bt_mesh_adv.adv_send: type 1 len 28: 37d544ee68f31aa29525ffd3c210726~                                                   
[00:00:01.072,937] <dbg> bt_mesh_adv.adv_send: count 2 interval 20ms duration 90ms                                                               
[00:00:01.073,577] <dbg> bt_mesh_lpn.lpn_set_state: enabled -> wait offer                                                                        
[00:00:01.073,577] <dbg> bt_mesh_adv.adv_send: Advertising started. Sleeping 90 ms                                                               
[00:00:01.163,848] <dbg> bt_mesh_adv.adv_send: Advertising stopped                                                                               
[00:00:01.719,512] <dbg> bt_mesh_lpn.bt_mesh_lpn_friend_offer: recv_win 255 queue_size 16 sub_list_size 3 rssi -40 counter 0                     
--- 1 messages dropped ---                                                                                                                       
[00:00:01.721,984] <dbg> bt_mesh_lpn.send_friend_poll: lpn->sent_req 0x00                                                                        
[00:00:01.722,106] <dbg> bt_mesh_adv.bt_mesh_adv_send: type 0x01 len 19: 6826064ea4a0815b994af614d212464~                                        
[00:00:01.722,198] <dbg> bt_mesh_adv.adv_send: type 1 len 19: 6826064ea4a0815b994af614d212464~                                                   
[00:00:01.722,198] <dbg> bt_mesh_adv.adv_send: count 2 interval 20ms duration 90ms                                                               
[00:00:01.722,839] <dbg> bt_mesh_lpn.req_sent: req 0x01 duration 90 err 0 state wait offer                                                       
[00:00:01.722,839] <dbg> bt_mesh_adv.adv_send: Advertising started. Sleeping 90 ms                                                               
[00:00:01.813,110] <dbg> bt_mesh_adv.adv_send: Advertising stopped                                                                               
[00:00:01.932,800] <dbg> bt_mesh_adv.bt_mesh_scan_cb: len 26: 192a6871a75b3159d1d16e55efcdb79~                                                   
[00:00:01.932,922] <inf> bt_mesh_lpn: Friendship established with 0x0001                                                                         
[00:00:01.932,922] <dbg> bt_mesh_lpn.friend_response_received: lpn->sent_req 0x01                                                                           
[00:00:01.932,952] <dbg> bt_mesh_adv.bt_mesh_scan_disable:                                                                                       
[00:00:01.933,197] <dbg> bt_mesh_lpn.lpn_set_state: wait offer -> established                                                                    
[00:00:01.933,197] <dbg> bt_mesh_lpn.bt_mesh_lpn_friend_update: flags 0x00 iv_index 0x00000000 md 0                                              
[00:00:01.933,197] <dbg> bt_mesh_lpn.sub_update: op 0x07 sent_req 0x00                                                                           
[00:00:01.933,349] <dbg> bt_mesh_adv.bt_mesh_adv_send: type 0x01 len 21: <log_strdup alloc failed>                                               
[00:00:01.933,380] <dbg> bt_mesh_lpn.sub_update: op 0x08 sent_req 0x07                                                                           
[00:00:01.933,410] <dbg> bt_mesh_adv.adv_send: type 1 len 21: <log_strdup alloc failed>                                                          
[00:00:01.933,441] <dbg> bt_mesh_adv.adv_send: count 2 interval 20ms duration 90ms                                                               
[00:00:01.933,959] <dbg> bt_mesh_lpn.req_sent: req 0x07 duration 90 err 0 state established                                                      
[00:00:01.933,959] <dbg> bt_mesh_lpn.lpn_set_state: established -> recv delay                                                                    
[00:00:01.933,990] <dbg> bt_mesh_adv.adv_send: Advertising started. Sleeping 90 ms                                                               
[00:00:02.024,078] <dbg> bt_mesh_lpn.lpn_timeout: state: recv delay                                                                              
[00:00:02.024,078] <dbg> bt_mesh_adv.bt_mesh_scan_enable:                                                                                        
[00:00:02.024,353] <dbg> bt_mesh_adv.adv_send: Advertising stopped                                                                               
[00:00:02.029,998] <err> bt_mesh_adv: starting scan failed (err -5)                                                                              
[00:00:02.029,998] <dbg> bt_mesh_lpn.lpn_set_state: recv delay -> wait update                                                                    
[00:00:02.379,150] <dbg> bt_mesh_lpn.lpn_timeout: state: wait update                                                                             
[00:00:02.379,180] <wrn> bt_mesh_lpn: No response from Friend during ReceiveWindow                                                               
[00:00:02.379,180] <dbg> bt_mesh_adv.bt_mesh_scan_disable:                                                                                       
[00:00:02.379,180] <dbg> bt_mesh_lpn.lpn_set_state: wait update -> established                                                                   
[00:00:02.479,248] <dbg> bt_mesh_lpn.lpn_timeout: state: established                                                                             
[00:00:02.479,248] <dbg> bt_mesh_lpn.sub_update: op 0x07 sent_req 0x00                                                                           
[00:00:02.479,400] <dbg> bt_mesh_adv.bt_mesh_adv_send: type 0x01 len 21: 68a123aa2fa3d96a54e52c0673a7248~                                        
[00:00:02.479,461] <dbg> bt_mesh_adv.adv_send: type 1 len 21: 68a123aa2fa3d96a54e52c0673a7248~                                                   
[00:00:02.479,461] <dbg> bt_mesh_adv.adv_send: count 2 interval 20ms duration 90ms                                                               
[00:00:02.480,010] <dbg> bt_mesh_lpn.req_sent: req 0x07 duration 90 err 0 state established                                                      
[00:00:02.480,041] <dbg> bt_mesh_lpn.lpn_set_state: established -> recv delay                                                                    
[00:00:02.480,041] <dbg> bt_mesh_adv.adv_send: Advertising started. Sleeping 90 ms                                                                                                       
[00:00:02.570,129] <dbg> bt_mesh_lpn.lpn_timeout: state: recv delay                                                                              
[00:00:02.570,129] <dbg> bt_mesh_adv.bt_mesh_scan_enable:                                                                                        
[00:00:02.570,404] <dbg> bt_mesh_adv.adv_send: Advertising stopped                                                                               
[00:00:02.576,049] <err> bt_mesh_adv: starting scan failed (err -5)                                                                              
[00:00:02.576,049] <dbg> bt_mesh_lpn.lpn_set_state: recv delay -> wait update                                                                    
[00:00:02.925,201] <dbg> bt_mesh_lpn.lpn_timeout: state: wait update                                                                             
[00:00:02.925,231] <wrn> bt_mesh_lpn: No response from Friend during ReceiveWindow                                                               
[00:00:02.925,231] <dbg> bt_mesh_adv.bt_mesh_scan_disable:                                                                                       
[00:00:02.925,231] <dbg> bt_mesh_lpn.lpn_set_state: wait update -> established                                                                   
[00:00:03.025,299] <dbg> bt_mesh_lpn.lpn_timeout: state: established                                                                             
[00:00:03.025,299] <dbg> bt_mesh_lpn.sub_update: op 0x07 sent_req 0x00                                                                           
[00:00:03.025,451] <dbg> bt_mesh_adv.bt_mesh_adv_send: type 0x01 len 21: 684b5758589ece2ea58eb80e6762f84~                                        
[00:00:03.025,512] <dbg> bt_mesh_adv.adv_send: type 1 len 21: 684b5758589ece2ea58eb80e6762f84~                                                   
[00:00:03.025,512] <dbg> bt_mesh_adv.adv_send: count 2 interval 20ms duration 90ms                                                               
[00:00:03.026,062] <dbg> bt_mesh_lpn.req_sent: req 0x07 duration 90 err 0 state established                                                      
[00:00:03.026,062] <dbg> bt_mesh_lpn.lpn_set_state: established -> recv delay                                                                    
[00:00:03.026,092] <dbg> bt_mesh_adv.adv_send: Advertising started. Sleeping 90 ms                                                                                                       
[00:00:03.116,180] <dbg> bt_mesh_lpn.lpn_timeout: state: recv delay                                                                              
[00:00:03.116,180] <dbg> bt_mesh_adv.bt_mesh_scan_enable:                                                                                        
[00:00:03.116,455] <dbg> bt_mesh_adv.adv_send: Advertising stopped                                                                               
[00:00:03.122,100] <err> bt_mesh_adv: starting scan failed (err -5)                                                                              
[00:00:03.122,100] <dbg> bt_mesh_lpn.lpn_set_state: recv delay -> wait update                                                                    
[00:00:03.471,252] <dbg> bt_mesh_lpn.lpn_timeout: state: wait update                                                                             
[00:00:03.471,282] <wrn> bt_mesh_lpn: No response from Friend during ReceiveWindow                                                               
[00:00:03.471,282] <dbg> bt_mesh_adv.bt_mesh_scan_disable:                                                                                       
[00:00:03.471,282] <dbg> bt_mesh_lpn.lpn_set_state: wait update -> established                                                                   
[00:00:03.571,350] <dbg> bt_mesh_lpn.lpn_timeout: state: established                                                                             
[00:00:03.571,350] <dbg> bt_mesh_lpn.sub_update: op 0x07 sent_req 0x00                                                                           
[00:00:03.571,502] <dbg> bt_mesh_adv.bt_mesh_adv_send: type 0x01 len 21: 6863b4482c9f145cf3a426b87b388fc~                                        
[00:00:03.571,563] <dbg> bt_mesh_adv.adv_send: type 1 len 21: 6863b4482c9f145cf3a426b87b388fc~                                                   
[00:00:03.571,563] <dbg> bt_mesh_adv.adv_send: count 2 interval 20ms duration 90ms                                                               
[00:00:03.572,113] <dbg> bt_mesh_lpn.req_sent: req 0x07 duration 90 err 0 state established                                                      
[00:00:03.572,143] <dbg> bt_mesh_lpn.lpn_set_state: established -> recv delay                                                                    
[00:00:03.572,143] <dbg> bt_mesh_adv.adv_send: Advertising started. Sleeping 90 ms                                                                                                        
[00:00:03.662,231] <dbg> bt_mesh_lpn.lpn_timeout: state: recv delay                                                                              
[00:00:03.662,231] <dbg> bt_mesh_adv.bt_mesh_scan_enable:                                                                                        
[00:00:03.662,506] <dbg> bt_mesh_adv.adv_send: Advertising stopped                                                                               
[00:00:03.668,151] <err> bt_mesh_adv: starting scan failed (err -5)                                                                              
[00:00:03.668,151] <dbg> bt_mesh_lpn.lpn_set_state: recv delay -> wait update                                                                    
[00:00:04.017,303] <dbg> bt_mesh_lpn.lpn_timeout: state: wait update                                                                             
[00:00:04.017,333] <wrn> bt_mesh_lpn: No response from Friend during ReceiveWindow                                                               
[00:00:04.017,333] <dbg> bt_mesh_adv.bt_mesh_scan_disable:                                                                                       
[00:00:04.017,333] <dbg> bt_mesh_lpn.lpn_set_state: wait update -> established                                                                   
[00:00:04.117,401] <dbg> bt_mesh_lpn.lpn_timeout: state: established                                                                             
[00:00:04.117,401] <err> bt_mesh_lpn: No response from Friend after 4 retries                                                                    
[00:00:04.117,401] <dbg> bt_mesh_lpn.clear_friendship: force 0 disable 0                                                                         
[00:00:04.117,431] <dbg> bt_mesh_lpn.send_friend_clear:                                                                                          
[00:00:04.117,553] <dbg> bt_mesh_adv.bt_mesh_adv_send: type 0x01 len 22: 37a1d5a5a62bf5bc5b4f6ab021e2f97~                                        
[00:00:04.117,614] <dbg> bt_mesh_adv.adv_send: type 1 len 22: 37a1d5a5a62bf5bc5b4f6ab021e2f97~                                                   
[00:00:04.117,645] <dbg> bt_mesh_adv.adv_send: count 3 interval 20ms duration 120ms                                                              
[00:00:04.118,194] <dbg> bt_mesh_adv.adv_send: Advertising started. Sleeping 120 ms                                                              
[00:00:04.238,494] <dbg> bt_mesh_adv.bt_mesh_scan_enable:                                                                                        
[00:00:04.238,830] <dbg> bt_mesh_lpn.lpn_set_state: established -> clear                                                                         
[00:00:04.238,861] <dbg> bt_mesh_adv.adv_send: Advertising stopped                                                                               
[00:00:05.338,928] <dbg> bt_mesh_lpn.lpn_timeout: state: clear                                                                                   
[00:00:05.338,958] <dbg> bt_mesh_lpn.clear_friendship: force 0 disable 0                                                                         
[00:00:05.338,958] <dbg> bt_mesh_lpn.send_friend_clear:                                                                                          
[00:00:05.339,080] <dbg> bt_mesh_adv.bt_mesh_adv_send: type 0x01 len 22: 3754d338cb5ea322ab687a2cce7ed7d~                                        
[00:00:05.339,141] <dbg> bt_mesh_adv.adv_send: type 1 len 22: 3754d338cb5ea322ab687a2cce7ed7d~                                                   
[00:00:05.339,172] <dbg> bt_mesh_adv.adv_send: count 3 interval 20ms duration 120ms                                                              
[00:00:05.339,782] <dbg> bt_mesh_adv.adv_send: Advertising started. Sleeping 120 ms                                                              
[00:00:05.398,559] <dbg> bt_mesh_adv.bt_mesh_scan_cb: len 24: 172a377b396324503cc9f2135618b54~                                                                                                                
[00:00:05.398,681] <wrn> bt_mesh_transport: Ignoring unexpected message in Low Power mode                                                        
[00:00:05.398,681] <wrn> bt_mesh_net: Removing rejected message from Network Message Cache                                                       
[00:00:05.422,546] <dbg> bt_mesh_adv.bt_mesh_scan_cb: len 24: 172a377b396324503cc9f2135618b54~                                                   
[00:00:05.447,143] <dbg> bt_mesh_adv.bt_mesh_scan_cb: len 24: 172a377b396324503cc9f2135618b54~                                                   
[00:00:05.460,021] <dbg> bt_mesh_adv.bt_mesh_scan_enable:                                                                                        
[00:00:05.460,906] <dbg> bt_mesh_lpn.lpn_set_state: clear -> clear                                                                             
[00:00:05.460,906] <dbg> bt_mesh_adv.adv_send: Advertising stopped                                                                               
[00:00:06.381,774] <dbg> bt_mesh_adv.bt_mesh_scan_cb: len 31: 1eff4c00121900efb45e5718bbad3e0~                                                   
[00:00:06.560,974] <dbg> bt_mesh_lpn.lpn_timeout: state: clear                                                                                   
[00:00:06.560,974] <dbg> bt_mesh_lpn.clear_friendship: force 0 disable 0                                                                         
[00:00:06.561,004] <dbg> bt_mesh_lpn.lpn_set_state: clear -> enabled                                                                             
[00:00:07.384,246] <dbg> bt_mesh_adv.bt_mesh_scan_cb: len 31: 1eff4c00121900efb45e5718bbad3e0~                                                   
[00:00:08.561,065] <dbg> bt_mesh_lpn.lpn_timeout: state: enabled                                                                                 
[00:00:08.561,096] <dbg> bt_mesh_lpn.send_friend_req:                                                                                            
[00:00:08.561,248] <dbg> bt_mesh_adv.bt_mesh_adv_send: type 0x01 len 28: 37477c9eda767ac90b7e12c81e9b436~                                        
[00:00:08.561,309] <dbg> bt_mesh_adv.adv_send: type 1 len 28: 37477c9eda767ac90b7e12c81e9b436~                                                   
[00:00:08.561,309] <dbg> bt_mesh_adv.adv_send: count 2 interval 20ms duration 90ms                                                               
[00:00:08.562,042] <dbg> bt_mesh_lpn.lpn_set_state: enabled -> wait offer                                                                        
[00:00:08.562,042] <dbg> bt_mesh_adv.adv_send: Advertising started. Sleeping 90 ms                                                               
[00:00:08.652,313] <dbg> bt_mesh_adv.adv_send: Advertising stopped                                                                               
[00:00:08.888,031] <dbg> bt_mesh_adv.bt_mesh_scan_cb: len 31: 1eff4c00121900efb45e5718bbad3e0~                                                   
[00:00:09.212,158] <dbg> bt_mesh_adv.bt_mesh_scan_cb: len 26: 192a37461ac77de88f099991f4088e5~                                                   
[00:00:09.212,280] <dbg> bt_mesh_lpn.bt_mesh_lpn_friend_offer: recv_win 255 queue_size 16 sub_list_size 3 rssi -39 counter 1                     
[00:00:09.214,782] <dbg> bt_mesh_lpn.send_friend_poll: lpn->sent_req 0x00                                                                        
[00:00:09.214,904] <dbg> bt_mesh_adv.bt_mesh_adv_send: type 0x01 len 19: <log_strdup alloc failed>                                               
[00:00:09.214,965] <dbg> bt_mesh_adv.adv_send: type 1 len 19: <log_strdup alloc failed>                                                          
[00:00:09.214,965] <dbg> bt_mesh_adv.adv_send: count 2 interval 20ms duration 90ms                                                               
[00:00:09.215,606] <dbg> bt_mesh_lpn.req_sent: req 0x01 duration 90 err 0 state wait offer                                                       
[00:00:09.215,637] <dbg> bt_mesh_adv.adv_send: Advertising started. Sleeping 90 ms                                                               
[00:00:09.305,847] <dbg> bt_mesh_adv.adv_send: Advertising stopped                                                                               
[00:00:09.415,008] <dbg> bt_mesh_adv.bt_mesh_scan_cb: len 26: 192a605c3bb3ae4dc1980dd4dd99b86~                                                   
[00:00:09.415,130] <inf> bt_mesh_lpn: Friendship established with 0x0001                                                                         
[00:00:09.415,161] <dbg> bt_mesh_lpn.friend_response_received: lpn->sent_req 0x01                                                                
[00:00:09.415,161] <dbg> bt_mesh_adv.bt_mesh_scan_disable:                                                                                       
[00:00:09.415,405] <dbg> bt_mesh_lpn.lpn_set_state: wait offer -> established                                                                    
```

**Environment (please complete the following information):**
 - Zephyr project in nRF52840 boards

**Additional context**
Tracking the error, it is generated, apparently, in hci_core.c, when the function "start_le_scan(u8_t scan_type, u16_t interval, u16_t window)"  uses the "err = set_random_address(&bt_dev.id_addr[0].a);" function.

### PR DESCRIPTION
**Pull request by trond-snekvik**
The passive scanner isn't actually using the advertising address, and
doesn't need to set it during scan initialization.

set_random_addr always fails if there's an active advertiser, which
would cause needless complications for the Bluetooth Mesh LPN case.

Fixes #22088.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

### CODE REVIEW

**Comment by joerchan** [comment]
This is a continuation of https://github.com/zephyrproject-rtos/zephyr/pull/21714

**Comment by jhedberg** [comment]
I might remember wrong, but I have a faint memory of us discussing this with the controller guys some time ago, and there was something that still justified setting the address for passive scanning. Perhaps it was directed advertising, or something similar, or then I just remember wrong and it was for something else. @cvinayak @carlescufi any comments?

**Comment by joerchan** [comment]
> I might remember wrong, but I have a faint memory of us discussing this with the controller guys some time ago, and there was something that still justified setting the address for passive scanning. Perhaps it was directed advertising, or something similar, or then I just remember wrong and it was for something else. @cvinayak @carlescufi any comments?

@jhedberg That's true. But then using advertiser with NRPA at the same time would not work. We could possibly set random address if the advertiser is not running. Otherwise we have a structural conflict with the passive scanner on using the random address as well.

**Comment by joerchan** [comment]
@jhedberg @carlescufi @trond-snekvik 
Maybe we could try to set the random address, but if it fails we continue without it (for passive scanner only). Then we would not get directed advertising reports. Then we could support the Mesh use-case of NRPA advertiser and Passive scanner.

Of course this should be documented clearly. It would only affect users who start advertiser with NRPA or an identity different than the default identity (BT_ID_DEFAULT).

**Comment by joerchan** [comment]
Furthermore. This should be documented, or possibly handled together with the `Central Address Resolution` characteristic.

**Comment by joerchan** [comment]
Current behaviour when Privacy is OFF and Random Static Identity

 - Active scanner sets an NRPA if not advertising or Kconfig option BT_SCAN_WITH_IDENTITY is set.
   Active scanner currently will not receive directed advertise reports.

 - Passive scanner sets random address.
   Passive scanner currently can received directed advertise reports.

 - Passive scanner with non-connectable advertiser
   Advertiser will overwrite random address of passive scanner, and will not receive directed advertise reports anymore.

 - Active scanner with non-connectable advertiser
   Both will try to set NRPA, and it will fail (because it is not the same NRPA)

Suggested action:
 - Switch default BT_SCAN_WITH_IDENTITY to on. (off by default) to support directed advertising reports.
 - Mesh sets BT_SCAN_WITH_IDENTITY to off
 - When BT_SCAN_WITH_IDENTITY is off, allow advertiser to overwrite the address NRPA of the scanner.
   - Should passive scanner set NRPA instead of Random Static Identity?
     It would be confusing that passive scanner receives directed advertising reports, but not active scanner

**Comment by carlescufi** [comment]
Conclusion after meeting:

- `BT_SCAN_WITH_IDENTITY` enables not only using (and setting) the identity address when scanning (and no privacy) but also the reception of directed adv reports for **both** active and passive scanning.
- Document this fact in both the Kconfig option help text and the `bt_le_scan_start()` doxygen

**Comment by trond-snekvik** [comment]
The mesh still works after @joerchan's changes.

**Review by jhedberg — APPROVED** [review]

File: subsys/bluetooth/host/hci_core.c [hunk 1]
```diff
@@ -695,6 +690,32 @@ static bool bt_le_adv_random_addr_check(const struct bt_le_adv_param *param)
...
+		 */
+
+		if (!(param->options & BT_LE_ADV_OPT_CONNECTABLE) &&
+		    (param->options & BT_LE_ADV_OPT_USE_IDENTITY)) {
+			/* Attempt to set non-connectable NRPA */
+			return false;
+		}
+
+
+		if (((param->options & BT_LE_ADV_OPT_CONNECTABLE) ||
```
Comment (cvinayak) [inline subsys/bluetooth/host/hci_core.c]: `((param->options & BT_LE_ADV_OPT_CONNECTABLE) || !(param->options & BT_LE_ADV_OPT_USE_IDENTITY))`

Is redundant as it evaluates to NOT operation of line 703, instead use  `else if` clause.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/host/hci_core.c (hunk 1 above)
Comment (joerchan) [reply subsys/bluetooth/host/hci_core.c]: Thanks. Updated.

**Review by cvinayak — APPROVED** [review]

**Pull request merged**